tests/abc_atlas_cache/test_cache.py

```diff
@@ -0,0 +1,573 @@
+            version=version,
+            test_bucket_name=self.test_bucket_name,
+            file_hash=true_checksum
+        )
+
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=metadata_path,
+                               Body=data)
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=data_path,
+                               Body=data)
+
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=self.new_manifest_string,
+                               Body=bytes(json.dumps(manifest), 'utf-8'))
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        cache.load_manifest(self.new_manifest_string)
+
+        expected_data_path = (self.cache_dir / data_path)
+        assert not expected_data_path.exists()
+        expected_metadata_path = (self.cache_dir / metadata_path)
+        assert not expected_metadata_path.exists()
+
+        # test data_path
+        directory = data_path.split('/')[1]
+        data_path_list = cache.download_directory_data(
+            directory=directory
+        )
+        assert data_path_list == [expected_data_path]
+        assert expected_data_path.exists()
+
+        expected_path = self.cache_dir / metadata_path
+        assert not expected_path.exists()
+
+        directory = metadata_path.split('/')[1]
+        metadata_path_list = cache.download_directory_metadata(
+            directory
+        )
+        assert metadata_path_list == [expected_path]
+        assert expected_path.exists()
+
+    def test_hashing_check(self):
+        """
+        Test that S3CloudCache.download_data() correctly downloads files from
+        S3
+        """
+        data = b'11235813kjlssergwesvsdd'
+
+        version = self.new_version
+        manifest, metadata_path, data_path = create_manifest_dict(
+            version=version,
+            test_bucket_name=self.test_bucket_name,
+            file_hash='junk'
+        )
+
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=metadata_path,
+                               Body=data)
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=data_path,
+                               Body=data)
+
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=self.new_manifest_string,
+                               Body=bytes(json.dumps(manifest), 'utf-8'))
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        cache.load_manifest(self.new_manifest_string)
+
+        expected_data_path = (self.cache_dir / data_path)
+        assert not expected_data_path.exists()
+        expected_metadata_path = (self.cache_dir / metadata_path)
+        assert not expected_metadata_path.exists()
+
+        # test data_path
```

Comment:
Can you flesh out the comment (or docstring) to explain that, the difference between the `download_data` calls that fail and those that succeed is `skip_hash_check`?